### PR TITLE
v.to.rast.multi: create raster maps for multiple attributes of a vector map

### PR DESCRIPTION
**Pull request by ninsbl**
This new module is a simple wrapper around v.to.rast to create raster maps for multiple attributes in an efficient way. Should save both time and storage.
It works with GRASS GIS 7 as well, so backporting might be appropriate....

### CODE REVIEW

File: src/vector/v.to.rast.multi/v.to.rast.multi.py [hunk 1]
```diff
@@ -0,0 +1,356 @@
...
+                                l=module_options["layer"],
+                                col=col,
+                            )
+                        )
+                    )
+                if at == "attribute_columns":
+                    if vmap_cols[col]["type"] not in supported_numeric:
+                        gscript.fatal(
+                            _(
+                                "Type {t} of column {col} is not a numeric type supportedby this module".format(
```
Comment (veroandreo) [inline src/vector/v.to.rast.multi/v.to.rast.multi.py:193]:
```suggestion
                                "Type {t} of column {col} is not a numeric type supported by this module".format(
```

**Comment by veroandreo** [comment]
I tried installing with g.extension, but fails:
```
g.extension extension=v.to.rast.multi url="https://github.com/ninsbl/grass-addons/" branch="v_to_rast_multi"
Fetching <v.to.rast.multi> from
<https://github.com/ninsbl/grass-addons/archive/v_to_rast_multi.zip> (be
patient)...
Compiling...
make: *** No targets specified and no makefile found.  Stop.
ERROR: Compilation failed, sorry. Please check above error messages.
```
I seem to recall there was a different way to install a PR aside from getting your branch locally. I cannot find it.

**Comment by veroandreo** [comment]
After getting the branch locally and installing the addon, I get the foolowing using the example from the manual:
```
v.to.rast.multi --o --v input=census_wake2000 type=area output=vrastmulti attribute_columns="RINGS_OK,TRACT" label_columns="ID,TRACTID" memory=3000 ndigits=",4" separator=","
Traceback (most recent call last):
  File "/home/veroandreo/.grass8/addons/scripts/v.to.rast.multi", line 356, in <module>
    sys.exit(main())
  File "/home/veroandreo/.grass8/addons/scripts/v.to.rast.multi", line 233, in main
    list(map(int, options["ndigits"].split(","))) if options["ndigits"] else None
ValueError: invalid literal for int() with base 10: ''
```
Seems there's an extra comma there... removing it, I get: 
```
v.to.rast.multi --o --v input=census_wake2000 type=area output=vrastmulti attribute_columns="RINGS_OK,TRACT" label_columns="ID,TRACTID" memory=3000 ndigits=4 separator=","
ERROR: If the <ndigits> option is used it has to have the same length
       (number of commas) as the <attribute_columns> option.
```
In this case RINGS_OK and TRACT columns are integer, so the example should then not use the `ndigits` option, IIUC, and indeed it works fine without that.

File: src/vector/v.to.rast.multi/v.to.rast.multi.html [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+
+This example is based on the North Carolina sample location (nc_spm_08_grass7) which
+can be downloaded from the
+<a href="https://grass.osgeo.org/download/data/">GRASS GIS website</a>.
+
+<div class="code"><pre>
+# Create raster maps for RINGS_OK amd TRACT attribute of the census_wake2000 map
+v.to.rast.multi --o --v input=census_wake2000 type=area \
+output=vrastmulti attribute_columns="RINGS_OK,TRACT" \
+label_columns="ID,TRACTID" memory=3000 ndigits=",4" separator=","
```
Comment (veroandreo) [inline src/vector/v.to.rast.multi/v.to.rast.multi.html:32]:
```suggestion
label_columns="ID,TRACTID" memory=3000 separator=","
```

File: src/vector/v.to.rast.multi/v.to.rast.multi.py (hunk 1 above)
Comment (ninsbl) [reply src/vector/v.to.rast.multi/v.to.rast.multi.py]: Thanks, addressed in 365732f

File: src/vector/v.to.rast.multi/v.to.rast.multi.html (hunk 2 above)
Comment (ninsbl) [reply src/vector/v.to.rast.multi/v.to.rast.multi.html:32]: This should work now. Unittests pass.

**Comment by veroandreo** [comment]
With the original command in the manual page I still get the same error(s) reported above. Only works when removing ndigits option completely.

File: src/vector/v.to.rast.multi/v.to.rast.multi.html (hunk 2 above)
Comment (veroandreo) [reply src/vector/v.to.rast.multi/v.to.rast.multi.html:32]: I get the exact same error as reported before. This conversation was resolved but which change fixed the error? I only see formatting and black commits after the error report, maybe you forgot? 
Also, I find it odd to express ndigits with a comma first. The definition says only integer, and only valid for floating point columns, which as reported earlier, RINGS_OK and TRACT are not. The example using ndigits won't work then.

**Comment by ninsbl** [comment]
Thanks, @veroandreo for looking at this again! You are right, I was not thinking staight...
I remembered incorrectly that I used the example in the testsuite (which passes). But in fact, there I used 0 instead of None (no number).
I updated the example and now also handle missing values (=0) in ndigits.

TRACT is according to `v.info -c census_wake2000` double precision... census_wake2000 may not be the best dataset as example here, but hopefully still good enough...

**Review by veroandreo — APPROVED** [review]
now yes :)

**Comment by ninsbl** [comment]
Thanks, @veroandreo for taking the time!

**Comment by ninsbl** [comment]
Backported in 78d7a9d

**Pull request merged**